_search-plugins/search-pipelines/ml-inference-search-response.md

```diff
@@ -388,4 +388,374 @@ The response confirms that the processor has generated text embeddings in the `p
+
+```json
+POST /_plugins/_ml/models/EOF6wJIBtDGAJRTD4kNg/_predict  
+{
+  "parameters": {
+    "prompt":"\n\nHuman: You are a professional data analysist. You will always answer question based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say I don't know. Context: ${parameters.context.toString()}. \n\n Human: please summarize the documents \n\n Assistant:",
+    "context":"Dr. Eric Goldberg is a fantastic doctor who has correctly diagnosed every issue that my wife and I have had. Unlike many of my past doctors, Dr. Goldberg is very accessible and we have been able to schedule appointments with him and his staff very quickly. We are happy to have him in the neighborhood and look forward to being his patients for many years to come."
+  }
+}
+```
+
+
+Step 1: Create a pipeline
+
+The following example shows you how to create a search pipeline for a generative AI model. The model requires a context field as input and generates a response. It summarizes the text in the review field and stores the summary in the ext.ml_inference.llm_response field of the search response.
+
+```json
+
+PUT /_search/pipeline/my_pipeline_request_review_llm
+{
+  "response_processors": [
+    {
+      "ml_inference": {
+        "tag": "ml_inference",
+        "description": "This processor is going to run llm",
+        "model_id": "EOF6wJIBtDGAJRTD4kNg",
+        "function_name": "REMOTE",
+        "input_map": [
+          {
+            "context": "review"
+          }
+        ],
+        "output_map": [
+          {
+            "ext.ml_inference.llm_response": "response"
+          }
+        ],
+        "model_config": {
+          "prompt": "\n\nHuman: You are a professional data analyst. You will always answer questions based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say I don't know. Context: ${parameters.context.toString()}. \n\n Human: please summarize the documents \n\n Assistant:"
+        },
+        "ignore_missing": false,
+        "ignore_failure": false
+      }
+    }
+  ]
+}
+
+```
+{% include copy-curl.html %}
+
+In this configuration:
+
+The model_id specifies the ID of the generative AI model.
+The function_name is set to "REMOTE", indicating an externally hosted model.
+The input_map maps the review field from the document to the context field expected by the model.
+The output_map specifies that the model's response should be stored in ext.ml_inference.llm_response in the search response.
+The model_config includes a prompt that instructs the model on how to process the input and generate a summary.
+
+Step 2: Index sample documents
+
+Index some sample documents to test the pipeline:
```

Comment:
How about change to bulk ? That will be easier, just one API call

Reply:
changed to bulk
```
POST /_bulk
{"index":{"_index":"review_string_index","_id":"1"}}
{"review":"Customer acquisition cost: January: $50, New customers: January: 500."}
{"index":{"_index":"review_string_index","_id":"2"}}
{"review":"Customer acquisition cost: February: $45, New customers: February: 600."}
{"index":{"_index":"review_string_index","_id":"3"}}
{"review":"Customer acquisition cost: March: $40, New customers: March: 750."}
```